### PR TITLE
TARDIS-3783-Analyze and Improve Short Polling Implementation

### PR DESCRIPTION
**Pull request by muhammadhamzakhan**

### CODE REVIEW

File: opsgenie_sdk/models/success_response.py [hunk 1]
```diff
@@ -77,6 +75,11 @@ def __init__(self, request_id=None, took=0.0, result=None, data=None, url=None):
         else:
             self.url = None
 
+        if api_client is not None:
```
Comment (bkaganyildiz) [inline opsgenie_sdk/models/success_response.py]: This seems like a bit unnecessary

Comment (muhammadhamzakhan) [reply opsgenie_sdk/models/success_response.py]: Agreed. I am gonna update the pull request

File: opsgenie_sdk/rest.py [hunk 2]
```diff
@@ -263,6 +263,7 @@ def request(self, method, url, query_params=None, headers=None,
             exception = exceptions.build_exception(response=r)
             raise exception
 
+        self.retries = -1
```
Comment (asqui) [inline opsgenie_sdk/rest.py]: Tracking retry count here seems a bit hacky and error-prone. You've fixed one problem here but I think there are more lurking. For example, what happens when all retries are exhausted? This method raises and doesn't reset retries to -1 (and it has no way to know that retries have been exhausted in order to do the reset). So I think `retry_count` is going to be wrong until after the next successful request?

Seem like retry counts should be maintained at the level where the retrying is being done, and passed in somehow?

File: opsgenie_sdk/models/success_response.py [hunk 3]
```diff
@@ -190,96 +186,67 @@ def url(self, url):
...
+                if hasattr(response.data, "alert_id"):
+                    self._id = response.data.alert_id
+                elif hasattr(response.data, "incident_id"):
+                    self._id = response.data.incident_id
             else:
+                should_retry = True
+
+            attempt_count += 1
+
+            if should_retry and attempt_count < MAX_NUMBER_OF_RETRIES:
```
[the remark this answers is not in the document]
Comment (muhammadhamzakhan) [reply opsgenie_sdk/models/success_response.py]: Good catch. I will fix it right away.

File: opsgenie_sdk/rest.py (hunk 2 above)
Comment (muhammadhamzakhan) [reply opsgenie_sdk/rest.py]: Yup, I agree. In this particular pull request, I was more focused towards improving the short polling implementation but then I realized that the `retry_count` wasn't being reset in the previous implementation. So, from the top of my head, I just reset it there. I am gonna dig a little deeper into it. Thanks for the heads up :))

Comment (muhammadhamzakhan) [reply opsgenie_sdk/rest.py]: Upon further investigation, I realized that this particular `retry_count` is only used for metrics' publishing and not for the actual retry policy of the SDK. The number of times the SDK retries an action is managed by the retry library and is independent of this variable. I have added this information to an already existent internal issue about improving the retry logic of the SDK and hopefully the implementation will be improved very soon

Comment (muhammadhamzakhan) [reply opsgenie_sdk/rest.py]: It's fixed in the following pull request: https://github.com/opsgenie/opsgenie-python-sdk/pull/17

**Pull request merged**